### PR TITLE
feat: add error message to retry modal

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
## **Description**
 - Modify retry modal to display error message
 - Add transaction exception type for cancel and speedup transaction
 - In Transaction component, when transactionObject has error, throw exception base on the cancel / speedup
 - In Transaction component, toggleRetry should take error message instead of error
 - In Transaction component, error message should pass into retry modal

## **Related issues**

Fixes: #

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by igorms-cons — DISMISSED** [review]

**Review by Cal-L — COMMENTED** [review]
left some comments

File: app/components/UI/Transactions/RetryModal/index.tsx [hunk 1]
```diff
@@ -32,16 +32,28 @@ const createStyles = (colors: any) =>
       textAlign: 'center',
       color: colors.text.default,
     },
+    modalErrText: {
+      ...(fontStyles.normal as any),
```
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply app/components/UI/Transactions/RetryModal/index.tsx:36]: any example?

[the remark this answers is not in the document]
Comment (dawnseeker8) [reply app/components/UI/Transactions/RetryModal/index.tsx:36]: hi, @Cal-L i didnt see any theme style in MM mobile code base, can you let me know where i can find those theme level global style,

[the remark this answers is not in the document]
Comment (Cal-L) [reply app/components/UI/Transactions/RetryModal/index.tsx:36]: Check out this file https://github.com/MetaMask/metamask-mobile/blob/main/app/components/UI/AccountApproval/styles.ts#L20. As for the style to apply, whoever provided the design should be using the correct design tokens provided in Figma. Upon inspection of Figma, you should see something such as `lBodyMD` being used as opposed to hardcoded styles.

[the remark this answers is not in the document]
Comment (dawnseeker8) [reply app/components/UI/Transactions/RetryModal/index.tsx]: have remove all fontStyle , fontSize , and use `Varient` attribute in <Text> component to use theme style.

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**